plugins/action_refresh_review_assignments/src/databaseMethods.ts

```diff
@@ -0,0 +1,58 @@
+const databaseMethods = (DBConnect: any) => ({
+  getApplicationsWithExistingReviewAssignments: async (userIds: number[]) => {
+    const text = `SELECT DISTINCT application_id
+      FROM review_assignment ra JOIN application app
+      ON ra.application_id = app.id
+      WHERE app.outcome = 'PENDING'
+      AND reviewer_id = ANY($1)
+    `
+    try {
+      const result = await DBConnect.query({ text, values: [userIds] })
+      return result.rows.map(({ application_id }: { application_id: number }) => application_id)
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  },
+  getApplicationsFromUserPermissions: async (userIds: number[]) => {
+    const text = `SELECT DISTINCT app.application_id
+      FROM permissions_all perm
+      JOIN application_stage_status_latest app
+      ON perm."templateId" = app.template_id
+      WHERE "userId" = ANY($1)
+      AND "permissionType" = ANY('{REVIEW, ASSIGN}')
+      AND app.outcome = 'PENDING'
+      AND app.status <> 'DRAFT'
```

Comment:
```suggestion
      AND app.status <> 'DRAFT'
```

Wondering about this, if the application is back for the Applicant to **make changes** then the application woudn't be listed as one of the ones required to update the reviewAssignment?

Reply:
Yeah, good point. Was just concerned about what would happen if you tried to generate review assignments for an application that had never been submitted yet. I think your action would create them, wouldn't it (which wouldn't be good)?

Reply:
@nmadruga see my suggestion for dealing with this: https://github.com/openmsupply/application-manager-server/pull/632#issuecomment-984213967